### ISSUE TITLE
In Salsah 1.5, it is currently impossible to search the content of a resource's label in advanced search

### ISSUE THREAD

**Issue by mrivoal** [dhlab-basel/Knora#567]
In Salsah 1.5, it is currently impossible to search the content of a resource's `label` property.

Because `label` doesn't appear on the list of avalaible properties for any resource within the vocabulary of a project.
![capture d ecran 2017-07-13 a 12 32 07](https://user-images.githubusercontent.com/17294309/28162625-0e3a6d40-67c8-11e7-8eb1-764ac6f3d5cc.png)

I thought it would be possible to search for a `label` within the Knora System Project vocabulary but it isn't because `label` isn't displayed here either.

The fulltext search however makes it possible to search within `labels`.  But I think it would make sense to make the `label` available as a property to search through for every resource in the advanced search window. What do you think?

**Comment by loicjaouen** [comment]
we could tweak the gui to add label by default, but what is the request that would look into `label`?

certainly not:
```
http://knora.unil.ch/v1/search/?searchtype=extended&property_id=http://www.w3.org/2000/01/rdf-schema#label&compop=MATCH&searchval=pictet&show_nrows=25&start_at=0&filter_by_restype=http://www.knora.org/ontology/0101#Institution
```

**Comment by benjamingeer** [comment]
You're right, it should be possible to do this, but it's not implemented in the Knora API server.

**Comment by benjamingeer** [comment]
Can this wait until Knora API v2? It will be much easier to implement it there.

**Comment by mrivoal** [comment]
I guess it depends. If you intend to release the Knora API v2 in the very next months and it is then possible to implement this functionality in Salsah 1.5, I think it can wait... 

But do you have any idea on when it could be implemented then?
And do you think it would be still be possible to implement it in Salsah 1.5?

**Comment by benjamingeer** [comment]
After talking it over with @tobiasschweizer, we think it will be easier and quicker to implement it in API v1 for you. Basically we need to treat the label as a pseudo-property in the GUI. @loicjaouen could you make the necessary changes in the GUI, and decide what should be submitted to the API and what the API should return? Then I think I can implement the necessary API functionality.

**Comment by loicjaouen** [comment]
Cool, thanks! I will propose something

**Comment by benjamingeer** [comment]
Does this still need to be done?

**Comment by loicjaouen** [comment]
sorry about that, yes, it still has to be done.

why not:

```
http://knora.unil.ch/v1/search/
  ?searchtype=extended
  &property_id=http://www.w3.org/2000/01/rdf-schema#label
  &compop=MATCH
  &searchval=pictet
  &show_nrows=25
  &start_at=0
  &filter_by_restype=http://www.knora.org/ontology/0101#Institution
```

or translated to `anything`:
```
http://localhost:3333/v1/search/
  ?searchtype=extended
  &property_id=http%3A%2F%2Fwww.w3.org%2Fontology%2Fanything%23hasDecimal
  &compop=EQ
  &searchval=Testding
  &show_nrows=25
  &start_at=0
  &filter_by_restype=http%3A%2F%2Fwww.knora.org%2Fontology%2Fanything%23Thing
  &filter_by_project=http%3A%2F%2Frdfh.ch%2Fprojects%2Fanything
```

**Comment by loicjaouen** [comment]
now `v2` is kind of out we have to balance the work between implementing it in `v2` client side (standardizing a request and adding a parser for `v2` answers) and adding a `v1` route on the back-end.
